src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -242,6 +244,18 @@ private ApiEvent InitializeApiEvent(string accountId)
             apiEvent.TokenType = AuthenticationRequestParameters.AuthenticationScheme.TelemetryTokenType;
             apiEvent.AssertionType = GetAssertionType();
 
+            if (AuthenticationRequestParameters.ExtraQueryParameters.ContainsKey("caller-sdk-id"))
```

Comment:
nit: You can use the `TryGet(key, out string val)` to combine ContainsKey and the [key] operations toghether.